src/lib/storage/storage_proposal.rb

```diff
@@ -0,0 +1,138 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2015] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "storage"
+require_relative "./disk_size"
+require_relative "./storage_manager"
+require_relative "./proposal_settings"
+require_relative "./proposal_volume"
+require_relative "./boot_requirements_checker"
+require_relative "./disk_analyzer"
+require_relative "./space_maker"
+require "pp"
+
+# This file can be invoked separately for minimal testing.
+# Use 'sudo' if you do that since it will do hardware probing with libstorage.
+
+module Yast
+  module Storage
+    #
+    # Storage proposal for installation: Class that can suggest how to create
+    # or change partitions for a Linux system installation based on available
+    # storage devices (disks) and certain configuration parameters.
+    #
+    class Proposal
+      include Yast::Logger
+
+      attr_accessor :settings
+
+      # devicegraph names
+      PROPOSAL = "proposal"
+      PROBED   = "probed"
+
+      def initialize
+        @settings = ProposalSettings.new
+        @proposal = nil # ::Storage::DeviceGraph
+        @disk_blacklist = []
+        @disk_greylist  = []
+      end
+
+      # Create a storage proposal.
+      def propose
+        storage = StorageManager.instance # this will start probing in the first invocation
```

Comment:
If somebody else already triggered the probing, that's fine. It will probe with the first usage, and that's just what is desired here. We certainly don't want to probe several times. This comment is supposed to mean "this will start probing unless somebody else already did that someplace else".